pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -220,6 +229,11 @@ public int filterEntriesForConsumer(Optional<MessageMetadata[]> optMetadataArray
 
         }
 
+        if (serviceConfig.isDispatchThrottlingForFilteredEntriesEnabled()) {
```

Comment:
one side effect of putting this here is that this affects the "analize-backlog" command.
also, we are blocking the dispatch thread.
the users will probably have to tune `numWorkerThreadsForNonPersistentTopic`, because the subscription dispatch threads will have some idle time due to this throttling

@MMirelli @michaeljmarshall @lhotari

Reply:
> one side effect of putting this here is that this affects the "analize-backlog" command.

Good point, I think that might be good though in order to prevent that command from consuming too many resources at a time in the broker.

> also, we are blocking the dispatch thread.

The call to `tryDispatchPermit` is non-blocking, other than needing to acquire a lock on the subscription's rate limiter. Here is the relevant code path:

https://github.com/apache/pulsar/blob/ad6213828c1f2e6b736657d3745522ac1e2b1bb1/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/DispatchRateLimiter.java#L100-L108

https://github.com/apache/pulsar/blob/84b65598481fd9bbb6e06e2deb335222a04b9c6b/pulsar-common/src/main/java/org/apache/pulsar/common/util/RateLimiter.java#L175-L202

It's interesting that the current solution acquires permits after sending messages. In looking at the `RateLimiter` code above, we completely ignore whether or not the specific messages actually get the permit to dispatch messages. This PR does not change those semantics though, so I think it is fine to leave it as is for this PR.

Reply:
I just noticed that `isDispatchOrPrecisePublishRateLimiter` is always true in this class, so there are no issues with getting the permits after delivery. We'll allow for some over-delivery, and then we will throttle consumers.